BedrockServer.cpp

```diff
@@ -538,18 +527,13 @@ void BedrockServer::sync()
                             SERROR("peekCommand (" << command->request.getVerb() << ") returned invalid result code: " << (int)result);
                         }
 
-                        // If we just started a new HTTPS request, save it for later.
+                        // If this command attempted an HTTP request, kill it.
                         if (command->httpsRequests.size()) {
-                            waitForHTTPS(move(command));
-                            // TODO:
-                            // Move the HTTPS loop into the worker, so that the worker can poll on its own requests.
-                            // This is the first step toward linear workers that run start->finish without being
-                            // interrupted and bounced back and forth between a bunch of queues.
-                            //
-                            // The follow-up to that is to allow direct escalations from workers to leader, which can
-                            // be done as a simple HTTPS request for the exact same command.
-
-                            // Move on to the next command until this one finishes.
+                            SWARN("Killing command " << command->request.methodLine << " that attempted HTTPS request in sync thread.");
+                            command->response.clear();
+                            command->response.methodLine = "500 Refused";
+                            command->complete = true;
+                            _reply(command);
```

Comment:
We ban the sync thread from doing HTTPS requests here. I could test this, but I don't think it will matter, no commands that do HTTPS requests should currently be able to make it to the sync thread, and we're trying to remove the remaining commands that could.

Reply:
NAB, we could add a log line instead of killing these commands if they exist for a short period of time (a day maybe) and then add this code. It might derisk this PR some.